app/models/template_input.rb

```diff
@@ -95,7 +103,13 @@ def ready?
     end
 
     def resolved_value
-      input_value.value
+      value = input_value.value
+
+      if @input.options_array.blank? || @input.options_array.include?(value)
+        value
+      else
+        raise(ValueNotPermitted, "Value '#{value}' is not a valid selection for #{@input.name}")
```

Comment:
Permission are evaluated when the run is scheduled (respecting view_hosts and view_job_templates permissions), until we add execute permission there are some gaps that will open more widely with delayed execution. We don't have designed permission granularity on input values, not sure if it's necessary. We can always hardcode parameters to templates if you want to limit user capabilities.